File: app/src/main/java/com/fastaccess/data/dao/LabelModel.java
package com.fastaccess.data.dao;

import android.os.Parcel;
import android.os.Parcelable;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Created by Kosh on 08 Dec 2016, 9:05 PM
 */

@Getter @Setter @NoArgsConstructor
public class LabelModel implements Parcelable {
    String url;
    String name;
    String color;

    @Override public int describeContents() { return 0; }

    @Override public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.url);
        dest.writeString(this.name);
        dest.writeString(this.color);
    }

    protected LabelModel(Parcel in) {
        this.url = in.readString();
        this.name = in.readString();
        this.color = in.readString();
    }

    public static final Creator<LabelModel> CREATOR = new Creator<LabelModel>() {
        @Override public LabelModel createFromParcel(Parcel source) {return new LabelModel(source);}

        @Override public LabelModel[] newArray(int size) {return new LabelModel[size];}
    };

    @Override public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        LabelModel that = (LabelModel) o;

        return name != null ? name.equals(that.name) : that.name == null;
    }

    @Override public int hashCode() {
        return name != null ? name.hashCode() : 0;
    }

    @Override public String toString() {
        return "LabelModel{" +
                "url='" + url + '\'' +
                ", name='" + name + '\'' +
                ", color='" + color + '\'' +
                '}';
    }
}

File: data/src/main/java/com/fastaccess/data/model/parcelable/LabelModel.kt
package com.fastaccess.data.model.parcelable

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class LabelModel(
    @SerializedName("name") var name: String? = null,
    @SerializedName("color") var color: String? = null,
    @SerializedName("url") var url: String? = null,
    @SerializedName("isDefault") var isDefault: Boolean? = null
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readValue(Boolean::class.java.classLoader) as? Boolean)

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeString(color)
        parcel.writeString(url)
        parcel.writeValue(isDefault)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<LabelModel> {
        override fun createFromParcel(parcel: Parcel): LabelModel {
            return LabelModel(parcel)
        }

        override fun newArray(size: Int): Array<LabelModel?> {
            return arrayOfNulls(size)
        }
    }
}